### ISSUE TITLE
Take advantage of format-preserving printer in phpstan/phpdoc-parser

### ISSUE THREAD

**Issue by ondrejmirtes** [rectorphp/rector#7960]
# Feature Request

Hello, I've added format-preserving pretty printer in phpdoc-parser 1.21.0. Details about how to use it are in the README: https://github.com/phpstan/phpdoc-parser#format-preserving-printer

I'm aware that you currently have your own solution, but maybe the built-in one will require less effort to make work, and you'll be able to delete a bunch of your own code.

Of course, it's not going to be have exactly 1:1 as your code does now, but we can work out the differences together for the best outcome 👍 I plan to support the format-preserving printer long into the future.

BTW: In the near future I want to add support for Doctrine annotations in phpdoc-parser and the printer. So you will need less hacks on your side to make them work too.

**Comment by yguedidi** [comment]
@ondrejmirtes maybe [this PR](https://github.com/rectorphp/rector-src/pull/3645) on rector-src will help when implementing support for Doctrine annotations
TL;DR: have a dedicated node for string type, when double quotes are used

**Comment by ondrejmirtes** [comment]
Phpdoc-parser already supports Doctrine snnotations as of 1.22.0 released last night. Rector should take advantage of that too 😊

**Comment by TomasVotruba** [comment]
Closing as we've implemented what we could re-use :+1:
